templates/cluster-template-flannel.yaml

```diff
@@ -0,0 +1,368 @@
+spec:
+  template:
+    spec:
+      kernel:
+        filename: boot/image
+        image: "${MVM_KERNEL_IMAGE:=ghcr.io/weaveworks-liquidmetal/flintlock-kernel:5.10.77}"
+      kernelCmdline: {}
+      memoryMb: 2048
+      networkInterfaces:
+      - guestDeviceName: eth1
+        type: macvtap
+      rootVolume:
+        id: root
+        image: "${MVM_ROOT_IMAGE:=ghcr.io/weaveworks-liquidmetal/capmvm-kubernetes:1.21.8}"
+      vcpu: 2
+---
+apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+kind: KubeadmConfigTemplate
+metadata:
+  name: "${CLUSTER_NAME}-md-0"
+spec:
+  template:
+    spec:
+      joinConfiguration:
+        nodeRegistration:
+          ignorePreflightErrors:
+          - SystemVerification
+          kubeletExtraArgs:
+            provider-id: "microvm://{{ ds.meta_data.vm_host }}/{{ ds.meta_data.instance_id }}"
+---
+apiVersion: addons.cluster.x-k8s.io/v1beta1
+kind: ClusterResourceSet
+metadata:
+  name: crs-flannel
+spec:
+  clusterSelector:
+    matchLabels:
+      cni: flannel
+  resources:
+  - kind: ConfigMap
+    name: flannel-addon
+  strategy: ApplyOnce
+---
+kind: ConfigMap
+metadata:
+  name: flannel-addon
+apiVersion: v1
+data:
+  flannel.yaml: |
+    ---
+    kind: Namespace
+    apiVersion: v1
+    metadata:
+      name: kube-flannel
+      labels:
+        pod-security.kubernetes.io/enforce: privileged
+    ---
+    kind: ClusterRole
+    apiVersion: rbac.authorization.k8s.io/v1
+    metadata:
+      name: flannel
+    rules:
+    - apiGroups:
+      - ""
+      resources:
+      - pods
+      verbs:
+      - get
+    - apiGroups:
+      - ""
+      resources:
+      - nodes
+      verbs:
+      - list
+      - watch
+    - apiGroups:
+      - ""
+      resources:
+      - nodes/status
+      verbs:
+      - patch
+    ---
+    kind: ClusterRoleBinding
+    apiVersion: rbac.authorization.k8s.io/v1
+    metadata:
+      name: flannel
+    roleRef:
+      apiGroup: rbac.authorization.k8s.io
+      kind: ClusterRole
+      name: flannel
+    subjects:
+    - kind: ServiceAccount
+      name: flannel
+      namespace: kube-flannel
+    ---
+    apiVersion: v1
+    kind: ServiceAccount
+    metadata:
+      name: flannel
+      namespace: kube-flannel
+    ---
+    kind: ConfigMap
+    apiVersion: v1
+    metadata:
+      name: kube-flannel-cfg
+      namespace: kube-flannel
+      labels:
+        tier: node
+        app: flannel
+    data:
+      cni-conf.json: |
+        {
+          "name": "cbr0",
+          "cniVersion": "0.3.1",
+          "plugins": [
+            {
+              "type": "flannel",
+              "delegate": {
+                "hairpinMode": true,
+                "isDefaultGateway": true
+              }
+            },
+            {
+              "type": "portmap",
+              "capabilities": {
+                "portMappings": true
+              }
+            }
+          ]
+        }
+      net-conf.json: |
+        {
+          "Network": "10.244.0.0/16",
```

Comment:
can this be defined the same way as `spec.clusterNetwork.pods.cidrBlocks`?

```
      - ${POD_CIDR:=10.244.0.0/16}
```

Reply:
lol it was totally meant to be. knew i had missed a bit

Reply:
nice spot ty 👍 fixed